packages/venia-concept/src/components/SwatchTooltip/swatchTooltip.js

```diff
@@ -0,0 +1,111 @@
+import React, { Component } from 'react';
+import { shape, string } from 'prop-types';
+import over from 'lodash.over';
+import uuid from 'uuid/v4';
+
+import classify from 'src/classify';
+import defaultClasses from './swatchTooltip.css';
+
+class SwatchTooltip extends Component {
+    static propTypes = {
+        classes: shape({
+            root: string,
+            tooltip: string
+        }),
+        text: string
+    };
+
+    constructor(...args) {
+        super(...args);
+
+        this.timeoutId = null;
+        this.uniqueId = uuid();
+    }
+
+    state = {
+        isShowing: false
+    };
+
+    /*
+     *  Close the tooltip on the next tick.
+     *  Unfortunately this is necessary because another child of ours may have received focus
+     *  and this blur event is fired before that new focus event.
+     *  @see https://reactjs.org/docs/accessibility.html#mouse-and-pointer-events.
+     */
+    onBlur = () => {
+        this.timeoutId = setTimeout(() => {
+            this.setState({ isShowing: false });
+        }, 0);
+    };
+
+    /*
+     *  The tooltip target has received focus, show the tooltip.
+     */
+    onFocus = () => {
+        // If a child receives focus, do not close the tooltip yet.
+        clearTimeout(this.timeoutId);
+
+        this.setState({ isShowing: true });
+    };
+
+    onKeyDown = event => {
+        if (event.key === 'Escape') {
```

Comment:
```suggestion
        if (event.key === 'Escape' || event.key === 'Tab') {
```

This would also hide it if you are moused in and then press tab. This could present an issue wherein there are focusable children within the button, imagine some text within the swatch, and pressing `Tab` should move focus to the text and keep the tooltip above the swatch.

Reply:
This suggestion would help solve the problem of multiple tooltips showing at the same time. This occurs when one Swatch has focus and another is being moused over.

But it may not be worth the effort.

1. We'd have to add logic to `SwatchTooltip` to workaround the new issue that it would present, which would be duplicating a lot of what `blur` is supposed to tell us.
1. Since this isn't a form, users usually aren't navigating with keyboard and mouse.

We could add some sort of `TooltipController` component that ensures that only one tooltip is showing at a time, but that seems like overkill too.

Reply:
Generally speaking, if the expected behavior is that only one tooltip can be open at a time, then it's not local state: it's shared state. So it sounds like this situation does call for a "controller". Context is good for this purpose.

Reply:
At my last company we had a `FocusManager` component that handled this sort of thing. Probably a bit overkill for now.

Reply:
@jimbo great points all around.

> if the expected behavior is that only one tooltip can be open at a time

This is the piece I'm iffy about. I mean, it sounds right but it is also a pretty rare occurrence so I lean towards "out of scope" here. When this thing makes it into Peregrine we can definitely generalize and make improvements to it.

What do you think?

Reply:
> When this thing makes it into Peregrine we can definitely generalize and make improvements to it.

Yeah, that's probably the right approach.